File: Easy/rank-transform-of-an-array.py
class Solution:     # 344ms
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        rank = {}
        a = []
        for i in sorted(arr):
            if i not in rank:
                rank[i] = len(rank) + 1
        for i in arr:
            a.append(rank[i])
                
        return a

File: Easy/rank-transform-of-an-array.java
class Solution{			// 75ms
	public int[] arrayRankTransform(int[] arr){
		HashMap<Integer, Integer> hm = new HashMap<>();
		int[] copy = arr.clone();
		Arrays.sort(copy);
		for(int x : copy){
			if(!hm.containsKey(x)){
				hm.put(x, hm.size() + 1);
			}
		}
		for(int i = 0; i < arr.length; i++){
			copy[i] = hm.get(arr[i]);
		}
		return copy;
	}
}